src/open_inwoner/accounts/tests/test_profile_views.py

```diff
@@ -541,52 +541,78 @@ def test_modify_phone_and_email_updates_klant_api(self, m):
         )
 
     @requests_mock.Mocker()
-    def test_eherkenning_user_updates_klant_api(self, m):
+    def test_eherkenning_user_updates_klant_api_with_rsin(self, m):
         MockAPIReadPatchData.setUpServices()
+        data = MockAPIReadPatchData()
 
-        for use_rsin_for_innNnpId_query_parameter in [True, False]:
-            with self.subTest(
-                use_rsin_for_innNnpId_query_parameter=use_rsin_for_innNnpId_query_parameter
-            ):
-                # NOTE Explicitly creating a new Mocker object here, because for some reason
-                # `m` is overridden somewhere, which causes issues when `MockAPIReadPatchData.install_mocks`
-                # is run for the second time
-                with requests_mock.Mocker() as m:
-                    data = MockAPIReadPatchData().install_mocks_eherkenning(
-                        m, use_rsin=use_rsin_for_innNnpId_query_parameter
-                    )
-
-                    config = OpenKlantConfig.get_solo()
-                    config.use_rsin_for_innNnpId_query_parameter = (
-                        use_rsin_for_innNnpId_query_parameter
-                    )
-                    config.save()
-
-                    response = self.app.get(self.url, user=data.eherkenning_user)
-
-                    # reset noise from signals
-                    m.reset_mock()
-                    self.clearTimelineLogs()
-
-                    form = response.forms["profile-edit"]
-                    form["email"] = "new@example.com"
-                    form["phonenumber"] = "0612345678"
-                    form.submit()
-
-                    # user data tested in other cases
-                    self.assertTrue(data.matchers[0].called)
-                    klant_patch_data = data.matchers[1].request_history[0].json()
-                    self.assertEqual(
-                        klant_patch_data,
-                        {
-                            "emailadres": "new@example.com",
-                            "telefoonnummer": "0612345678",
-                        },
-                    )
-                    self.assertTimelineLog("retrieved klant for user")
-                    self.assertTimelineLog(
-                        "patched klant from user profile edit with fields: emailadres, telefoonnummer"
-                    )
+        matchers = data.install_mocks_eherkenning(m, use_rsin=True).matchers
+
+        config = OpenKlantConfig.get_solo()
+        config.use_rsin_for_innNnpId_query_parameter = True
+        config.save()
+
+        response = self.app.get(self.url, user=data.eherkenning_user)
+
+        # reset noise from signals
+        m.reset_mock()
+        self.clearTimelineLogs()
```

Comment:
I removed the `reset_mock`, but `clearTimelineLogs` is actually still needed (it's also used in other tests in this file that don't have subtests)

Reply:
Existing code may or may not be perfect.

Could be cool to see why it happens but if you tested it fails if we remove it I'll believe it (could be because the code earlier in the tests generates similar logs)